internal/provider/generate.go

```diff
@@ -498,13 +505,24 @@ provider %[1]q {
 		return nil, err
 	}
 
-	g.infof("getting Terraform binary")
-	installer := &releases.ExactVersion{
-		Product:    product.Terraform,
-		Version:    version.Must(version.NewVersion("1.0.5")),
-		InstallDir: tmpDir,
+	i := install.NewInstaller()
+	src := []src.Source{}
+
+	if g.offlineMode {
+		g.infof("use Terraform binary from PATH")
+		src = append(src, &fs.AnyVersion{
+			Product: &product.Terraform,
+		})
+	} else {
+		g.infof("getting Terraform binary")
+		src = append(src, &releases.ExactVersion{
+			Product:    product.Terraform,
+			Version:    version.Must(version.NewVersion(defaultTerraformVersion)),
+			InstallDir: tmpDir,
+		})
```

Comment:
Recommend switching this to [`checkpoint.LatestVersion`](https://pkg.go.dev/github.com/hashicorp/hc-install@v0.3.1/checkpoint#LatestVersion), instead of keeping the hardcoded Terraform version.